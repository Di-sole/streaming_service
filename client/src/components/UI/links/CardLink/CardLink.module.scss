.container {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 0 12px 24px;
    padding: 12px;
    min-width: 153.14px;
    width: 154px;
    height: 235.05px;
    border-radius: 8px;

    background-color: #1f1b2e;
    transition-duration: 0.4s;

    &:hover {
        transform-origin: bottom;
        transform: scale(103%, 103%);

        .container__content {
            color: #fff;
        }
    }

    &__content {
        color: #a5a1b2;
        font-size: 15px;
        line-height: 20px;
        font-weight: 500;

        text-align: center;
    }
}

@media (max-width: 500px) {
    .container {
        min-width: 100px;
        width: 100px;
        height: 180px;
    }
}
