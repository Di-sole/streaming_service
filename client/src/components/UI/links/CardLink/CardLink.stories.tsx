import type { Meta, StoryObj } from '@storybook/react';
import Link from './CardLink';
import './CardLink.module.scss';

const meta: Meta<typeof Link> = {
    title: 'Links/CardLink',
    component: Link,
    parameters: {
        backgrounds: { default: 'dark' },
    },
};

export default meta;

type Story = StoryObj<typeof Link>;

export const CardLink: Story = {
    args: {
        href: '',
        children: 'Посмотреть все',
    },
};
