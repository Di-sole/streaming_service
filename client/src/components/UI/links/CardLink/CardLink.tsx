import Link from 'next/link';
import styles from './CardLink.module.scss';
import { ReactNode } from 'react';

interface ICardLinkProps {
    href: string;
    children: ReactNode;
    onClick?: () => void;
}

const CardLink = ({ href, children, onClick }: ICardLinkProps) => {
    return (
        <div onClick={onClick}>
            <Link href={href} className={styles.container}>
                <div className={styles.container__content}>{children}</div>
            </Link>
        </div>
    );
};

export default CardLink;
