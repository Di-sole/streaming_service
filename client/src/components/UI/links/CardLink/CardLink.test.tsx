import { render, screen } from '@testing-library/react';
import CardLink from './CardLink';

describe('CARDLINK SNAPSHOTS TESTS', () => {
    test('CardLink should not be changed', () => {
        render(<CardLink href="/" children="text" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
