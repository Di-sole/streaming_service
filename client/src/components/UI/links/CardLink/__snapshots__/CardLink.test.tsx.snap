// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CARDLINK SNAPSHOTS TESTS CardLink should not be changed 1`] = `
<a
  class="container"
  href="/"
>
  <div
    class="container__content"
  >
    text
  </div>
</a>
`;
